assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js

```diff
@@ -57,9 +153,14 @@ const WidgetAreaRenderer = ( { slug } ) => {
 			</div>
 			<div className="googlesitekit-widget-area-widgets">
 				<div className="mdc-layout-grid__inner">
-					{ widgets.map( ( widget ) => {
+					{ widgets.map( ( widget, i ) => {
 						return (
-							<Widget key={ widget.slug } slug={ widget.slug } />
+							<WidgetRenderer
+								className={ widgetClassNames[ i ] !== null ? classnames( widgetClassNames[ i ] ) : 'googlesitekit-widget-area--hidden' }
+								key={ widget.slug }
+								ref={ setRef( widget.slug ) }
+								slug={ widget.slug }
+							/>
 						);
 					} ) }
```

Comment:
This way of rendering widgets is only correct for the "boxes" style. For the "composite" style, this widgets loop should be wrapped in another:
```
<div className="mdc-layout-grid__cell mdc-layout-grid__cell--span-12">
    <div className="mdc-layout-grid">
        <div className="mdc-layout-grid__inner">
            { widgets }
        </div>
    </div>
</div>
```

See "Overall widget area grid and styling" in IB.